force-app/main/default/classes/SOQL.cls

```diff
@@ -288,21 +421,34 @@ public with sharing class SOQL {
      */
     public static List<Account> findAccountsByLeadEmailDomains(Set<Id> leadIds) {
         // Query the Leads
-
+        List<Lead> leads = [SELECT Email 
+                            FROM Lead 
+                            WHERE Id IN :leadIds];
+    
         // Create a set of email domains from the Leads
-
+        Set<String> emailDomains = new Set<String>();
+    
         // Loop through each Lead
-
-            // Get the email from the Lead
-
-            // Check if the email is not null
-
-                // Add the email domain to the set
-
-        
-        // Query the Accounts
-        return null; // Replace null with the variable you used to store the result
+        for (Lead leadRecord : leads) {
+            if (leadRecord.Email != null && leadRecord.Email.contains('@')) {
+                // Get the email domain
+                String emailDomain = leadRecord.Email.split('@')[1];
```

Comment:
In the future, consider using the **[substringAfter()](https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_string.htm#apex_System_String_substringAfter)** method.